extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/statistics/DistinctKeyCollector.java

```diff
@@ -43,8 +43,8 @@
  */
 public class DistinctKeyCollector implements KeyCollector<DistinctKeyCollector>
 {
-  static final int INITIAL_MAX_KEYS = 2 << 15 /* 65,536 */;
-  static final int SMALLEST_MAX_KEYS = 16;
+  static final int INITIAL_MAX_BYTES = 5_120_000;
+  static final int SMALLEST_MAX_BYTES = 5000;
```

Comment:
I'm not sure if these are the best thresholds here and for PARTITION_STATS_MAX_BYTES.